src/resdk/uploader.py

```diff
@@ -0,0 +1,215 @@
+"""The uploader class.
+
+Supports uploading to S3 bucket and to Genialis server.
+"""
+
+import uuid
+from enum import Enum, auto
+from pathlib import Path
+from typing import TYPE_CHECKING, Dict, Optional, Union
+from urllib.parse import urljoin
+
+import boto3
+import botocore
+
+from resdk.aws.session import get_refreshable_boto3_session
+from resdk.constants import CHUNK_SIZE
+from resdk.exceptions import ResolweServerError
+
+if TYPE_CHECKING:
+    from resdk.resolwe import Resolwe
+
+
+class UploadType(Enum):
+    """Supported uploader types."""
+
+    LOCAL = auto()
+    S3 = auto()
+
+    @staticmethod
+    def default():
+        """Return default upload type."""
+        return UploadType.LOCAL
+
+
+class Uploader:
+    """The uploader class.
+
+    It is used to upload the file to the Genialis platform. Is supports
+    uploading files to the server and also directly to the S3 bucket.
+    """
+
+    def __init__(self, resolwe: "Resolwe"):
+        """Store the ."""
+        self.resolwe = resolwe
+        self.upload_methods = {
+            UploadType.LOCAL: self._upload_local,
+            UploadType.S3: self._upload_s3,
+        }
+        # Use the following upload type if one can not be determined.
+        self._default_upload_type = UploadType.LOCAL
```

Comment:
Maybe rather `UploadType.default()`?

Reply:
I have removed this line. It is not needed anymore since the default has been moved to the enum.